### ISSUE TITLE
Significant slowdown using minify_html on large project

### ISSUE THREAD

**Issue by vallode** [Keats/gutenberg#2031]
# Bug Report

## Environment

Zola version: 0.16.1

## Expected Behavior
`minify_html = true` should add a non-significant amount of processing time to `zola build` and `zola serve`

## Current Behavior
`minify_html = true` adds around 2 seconds (sometimes much worse) to a 2 second build, doubling the build time.

Without `minify_html`
```
TERA_PRODUCTION=true zola build
Building site...
Checking all internal links with anchors.
> Successfully checked 0 internal link(s) with anchors.
-> Creating 3813 pages (0 orphan) and 44 sections
Done in 2.6s.
```

With `minify_html`
```
TERA_PRODUCTION=true zola build
Building site...
Checking all internal links with anchors.
> Successfully checked 0 internal link(s) with anchors.
-> Creating 3813 pages (0 orphan) and 44 sections
Done in 7.8s.
```

minify_html `0.10.3` CLI using ZSH `time` command
```
me# time ./minify-html --keep-closing-tags --minify-js --minify-css public/**/*.html 1>/dev/null

================
CPU	334%
user	2.431
system	0.464
total	0.866 
```

## Step to reproduce

1) `zola build` without minify_html
2) `zola build` with minify_html
3) Compare output time

Seemingly Zola runs the minification per file as it outputs it, instead of batching the minification, resulting in a large file/page count slowing the build process significantly.

**Comment by Keats** [comment]
`minify_html` is always called in parallel and there are no other API in https://docs.rs/minify-html/latest/minify_html/

I'm not really sure what to change there

**Comment by vallode** [comment]
When I initially had a look, I thought Zola is running minify_html _per file_ when it could realistically run it for _all files_ at the end of the build process. It's what I began doing on my own and it reduced the time of the build process a good few seconds on the large website I build.

https://github.com/getzola/zola/blob/26026a8a6452f5a14454e907668b2893cc10a4d0/components/site/src/lib.rs#L625-L632

**Comment by Keats** [comment]
> When I initially had a look, I thought Zola is running minify_html per file when it could realistically run it for all files at the end of the build process.

The lib api only takes a slice of bytes, there are no batch operations or using a glob.

**Comment by vallode** [comment]
I understand now. I guess the difference in speed was using a batch job on the CLI implementation of minify-html. Should we leave the issue open and I can open a separate one for a batch job implementation on their side?

**Comment by vallode** [comment]
According to https://github.com/wilsonzlin/minify-html/issues/126#issuecomment-1375445980 there are no implementation differences in the CLI, and since Zola is already running minify-html in parallel I'm at a loss for what could be causing the time difference in my builds. I'll attempt to create a minimal repo to demonstrate the slowdown but I also appreciate this isn't a major issue so there is no rush.

**Comment by EvanCarroll** [comment]
It would have been _really_ nice to have a bench marking suite created for this, or something to help us test a fix. You have a special use case, clearly. It's hard to act on it without being able to see it.

* -> Creating 3813 pages (0 orphan) and 44 sections

Without more information and with no way to a/b this, I vote close it.
